src/test/java/org/opensearch/security/test/helper/cluster/ClusterHelper.java

```diff
@@ -171,9 +174,15 @@ public final synchronized ClusterInfo startCluster(final NodeSettingsSupplier no
         for (int i = 0; i < internalClusterManagerNodeSettings.size(); i++) {
             NodeSettings setting = internalClusterManagerNodeSettings.get(i);
             int nodeNum = nodeNumCounter--;
-            PluginAwareNode node = new PluginAwareNode(setting.clusterManagerNode,
-                    getMinimumNonSecurityNodeSettingsBuilder(nodeNum, setting.clusterManagerNode, setting.dataNode, internalNodeSettings.size(), tcpClusterManagerPortsOnly, tcpPortsAllIt.next(), httpPortsIt.next())
-                            .put(nodeSettingsSupplier == null ? Settings.Builder.EMPTY_SETTINGS : nodeSettingsSupplier.get(nodeNum)).build(), setting.getPlugins());
+            final Settings.Builder nodeSettingsBuilder = getMinimumNonSecurityNodeSettingsBuilder(nodeNum, setting.clusterManagerNode, setting.dataNode, internalNodeSettings.size(), tcpClusterManagerPortsOnly, tcpPortsAllIt.next(), httpPortsIt.next());
```

Comment:
This is kind of painful - see AbstractSecurityUnitTest.mergeNodeRolesAndSettings where more background is provided.

Also, why not make this a function?  This code uses a lot of locals to pass information around and the method signature would be untenable.  Rather than refactor ClusterHelper for this case, I'm continuing the rich tradition of copy-pasting the update.

If we have strong feelings this could be an issue, but I doubt it would survive triage with the new test framework incoming

Reply:
If this a one-place-only kind of usage. I'm not against this approach.